pkg/kclient/port_forwarding.go

```diff
@@ -9,7 +9,10 @@ import (
 	"k8s.io/client-go/transport/spdy"
 )
 
-func (c *Client) SetupPortForwarding(pod *corev1.Pod, portPairs []string, out io.Writer, errOut io.Writer, stopChan chan struct{}) error {
+func (c *Client) SetupPortForwarding(pod *corev1.Pod, portPairs []string, out io.Writer, errOut io.Writer, stopChan chan struct{}, address string) error {
+	if address == "" {
+		address = "localhost"
```

Comment:
Kubernetes client considers `127.0.0.1` or `::1` as the localhost, and so I figured it would be safe to go with that.
https://github.com/redhat-developer/odo/blob/93d1330182e738888e7cbcffa904250239a83189/vendor/k8s.io/client-go/tools/portforward/portforward.go#L121-L133

But if you think it would help to be more explicit, I can do that.

Reply:
Oh ok, in this case, it should be fine then, as the translation of `localhost` into `127.0.0.1` is done in the `client-go` library itself, before it reaches the system for DNS resolution.
My concern was mostly about the different default values (`127.0.0.1` on Podman vs `localhost` on K8s), and checking if the port is free using `localhost`, which may actually return an unexpected result if `localhost` maps to a different loopback IP on the system.
I think it would make things consistent at least by using the same default `127.0.0.1` in all cases (when we listen and when we check if ports are available).
But as you want, [`9f947d2` (#6766)](https://github.com/redhat-developer/odo/pull/6766/commits/9f947d252a71b2d568f574b1d03e9eb208475a64) should be enough then to me.